hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -186,6 +187,16 @@ public String getString(String columnLabel) {
 
 	@Override
 	public boolean getBoolean(String columnLabel) {
+		final Object value = row.getValue( columnLabel );
+		wasNull = value == null;
+		if ( wasNull ) {
+			return false;
+		}
+
+		if( value instanceof BigInteger ) {
+			return ((BigInteger)value).intValue() > 0;
+		}
+
 		Boolean bool = row.getBoolean( columnLabel );
 		wasNull = bool == null;
```

Comment:
You don't need to reassign the wasnull. If it wa null before for `getValue`, it is still null for `geBoolean`